[DesktopVK] macos crashes when supplying Texture2DArray to shader with error `newSliceRange.length(3) must equal (1) for textureType (MTLTextureType2D).`

### Prerequisites

- [x] I have verified this issue is present in the `develop` branch
- [x] I have searched [open](https://github.com/MonoGame/MonoGame/issues) and [closed](https://github.com/MonoGame/MonoGame/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported.

### MonoGame Version

3.8.5-preview.3

### Which MonoGame platform are you using?

N/A

### Operating System

MacOS 26.2

### Description

Texture2D arrays on the Native/Vulkan backend create image views as VK_IMAGE_VIEW_TYPE_2D even when arrayLayers > 1. On macOS (MoltenVK/Metal), this results in a Metal assertion about slice range length for MTLTextureType2D.

I have a [fork and a commit](https://github.com/MonoGame/MonoGame/commit/24c426c744320b15fc486169da042955a2008e20) that appears to fix the problem if this issue is accepted.  Building MG locally and linking to the output appears to solve this problem.

### Steps to Reproduce

1. Use MonoGame Vulkan backend on macOS (Apple Silicon).
2. Create a Texture2D array (arraySize > 1) and sample it in a shader (Texture2DArray).
3. Run: app crashes with Metal assertion above.

### Minimal Example Repo

https://github.com/acarteas/mg-vulkan-texture-array-bug-poc

### Expected Behavior

The texture array should be sent to the shader as a texture array.

### Resulting Behavior

Console output when running app in POC repo:

```
dotnet run
/.../mg-vulkan-texture-array-bug-poc/example.Content/example.Content.csproj : warning NU1903: Package 'Microsoft.NETCore.App' 1.0.5 has a known high severity vulnerability, https://github.com/advisories/GHSA-7mfr-774f-w5r9
/.../mg-vulkan-texture-array-bug-poc/example.Content/example.Content.csproj : warning NU1903: Package 'Microsoft.NETCore.App' 1.0.5 has a known high severity vulnerability, https://github.com/advisories/GHSA-8884-xcr4-r68p
/.../mg-vulkan-texture-array-bug-poc/example.Content/example.Content.csproj : warning NU1903: Package 'Microsoft.NETCore.Jit' 1.0.7 has a known high severity vulnerability, https://github.com/advisories/GHSA-xcvr-qv8h-m7xw
Vulkan instance version: 1.2.296
Selected GPU: Apple M2 Pro
Supported Vulkan API version: 1.2.296
VK_EXT_custom_border_color is not supported by this driver!
VK_GOOGLE_hlsl_functionality1 is not supported by this driver!
VK_GOOGLE_user_type is not supported by this driver!
[mvk-error] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_D24_UNORM_S8_UINT is not supported on this device. Using VkFormat VK_FORMAT_D32_SFLOAT_S8_UINT instead.
Warning: Potentially unhandled layout transition from 5 to 6 in vulkan/MGG_Vulkan.cpp:2590
Warning: Potentially unhandled layout transition from 5 to 6 in vulkan/MGG_Vulkan.cpp:2590
Warning: Potentially unhandled layout transition from 5 to 7 in vulkan/MGG_Vulkan.cpp:2590
Warning: Potentially unhandled layout transition from 5 to 6 in vulkan/MGG_Vulkan.cpp:2590
Warning: Potentially unhandled layout transition from 5 to 7 in vulkan/MGG_Vulkan.cpp:2590
_mtlValidateArgumentsForTextureViewOnDevice:1866: failed assertion `Texture Creation
newSliceRange.length(3) must equal (1) for textureType (MTLTextureType2D).
'
```

### Files

_No response_

--- a/native/monogame/vulkan/MGG_Vulkan.cpp
+++ b/native/monogame/vulkan/MGG_Vulkan.cpp
@@ -636,12 +636,12 @@ static VkPrimitiveTopology ToVkPrimitiveTopology(MGPrimitiveType type)
 	}
 }
 
-static VkImageViewType ToVkImageViewType(MGTextureType type)
+static VkImageViewType ToVkImageViewType(MGTextureType type, int layerCount = 1)
 {
 	switch (type)
 	{
 	case MGTextureType::_2D:
-		return VK_IMAGE_VIEW_TYPE_2D;
+		return layerCount > 1 ? VK_IMAGE_VIEW_TYPE_2D_ARRAY : VK_IMAGE_VIEW_TYPE_2D;
 	case MGTextureType::_3D:
 		return VK_IMAGE_VIEW_TYPE_3D;
 	case MGTextureType::Cube:
@@ -1056,7 +1056,7 @@ static VkImageView CreateImageView(MGG_GraphicsDevice* device, MGG_Texture* text
 
 	VkImageViewCreateInfo image_view_create_info = { VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO };
 	image_view_create_info.image = texture->image;
-	image_view_create_info.viewType = ToVkImageViewType(texture->type);
+	image_view_create_info.viewType = ToVkImageViewType(texture->type, layer_count);
 	image_view_create_info.format = format;
 	image_view_create_info.subresourceRange.aspectMask = aspect_mask;
 	image_view_create_info.subresourceRange.baseMipLevel = 0;

